### PR TITLE
*: add doc for system variable tidb_idle_transaction_timeout

### PR DESCRIPTION
**Pull request by crazycs520**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

TiDB v7.6 introduces a new system variable: tidb_idle_transaction_timeout

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- Other reference link(s): https://github.com/pingcap/tidb/pull/48714

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -2666,6 +2666,17 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+### `tidb_idle_transaction_timeout` <span class="version-mark">从 v7.6.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 31536000]`
+- 单位：秒
+- 这个变量用来控制用户会话中事务的空闲超时。当用户会话处于事务状态中，且空闲时间超过该变量的值后，会话会被 Kill 掉。`0` 代表没有时间限制。
```
Comment (cfzjywxk) [inline system-variables.md:2678]: 这里需要解释下“空闲”状态的具体含义

Comment (crazycs520) [reply system-variables.md]: done

**Review by cfzjywxk — APPROVED** [review]

File: system-variables.md [hunk 2]
```diff
@@ -2666,6 +2666,18 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 31536000]`
+- 单位：秒
+- 这个变量用来控制用户会话中事务的空闲超时。当用户会话处于事务状态中，且空闲时间超过该变量的值后，会话会被 Kill 掉。用户会话空闲是指没有正在执行的请求，处于等待请求的状态。
+- 默认值 `0` 代表没有时间限制。
```
Comment (Oreoxmt) [inline system-variables.md:2679]:
```suggestion
- 默认值 `0` 表示没有时间限制。
```

File: system-variables.md [hunk 3]
```diff
@@ -2666,6 +2666,18 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+### `tidb_idle_transaction_timeout` <span class="version-mark">从 v7.6.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 31536000]`
+- 单位：秒
+- 这个变量用来控制用户会话中事务的空闲超时。当用户会话处于事务状态中，且空闲时间超过该变量的值后，会话会被 Kill 掉。用户会话空闲是指没有正在执行的请求，处于等待请求的状态。
```
Comment (Oreoxmt) [inline system-variables.md:2678]:
```suggestion
- 这个变量用来控制用户会话中事务的空闲超时。当用户会话处于事务状态且空闲时间超过该变量设定的值时，会话会被 Kill 掉。用户会话空闲是指没有正在执行的请求，处于等待请求的状态。
```

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 4]
```diff
@@ -2666,6 +2666,18 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 默认值：`168h`，即 7 天
 - 这个变量用来控制历史统计信息在存储中的保留时间。
 
+### `tidb_idle_transaction_timeout` <span class="version-mark">从 v7.6.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline system-variables.md:2669]: To enhance user understanding and provide comprehensive guidance, could we consider adding the description of this system variable in the following documents?

https://github.com/pingcap/docs-cn/blob/7c86675626516d5bce7ddc210d183e9a0db6ce63/faq/manage-cluster-faq.md?plain=1#L64-L66

https://github.com/pingcap/docs-cn/blob/7c86675626516d5bce7ddc210d183e9a0db6ce63/troubleshoot-tidb-oom.md?plain=1#L123-L131

https://github.com/pingcap/docs-cn/blob/7c86675626516d5bce7ddc210d183e9a0db6ce63/develop/dev-guide-timeouts-in-tidb.md?plain=1#L7-L9

https://github.com/pingcap/docs-cn/blob/7c86675626516d5bce7ddc210d183e9a0db6ce63/develop/dev-guide-connection-parameters.md?plain=1#L238-L240

Comment (crazycs520) [reply system-variables.md:2669]: @easonn7 PTAL

Comment (crazycs520) [reply system-variables.md:2669]: It looks like these related documents need to be updated？

Comment (crazycs520) [reply system-variables.md:2669]: Nice catch, done.

File: faq/manage-cluster-faq.md [hunk 5]
```diff
@@ -63,7 +63,7 @@ mysql -h 127.0.0.1 -uroot -P4000
 
 ### TiDB 是否支持会话超时？
 
-TiDB 目前支持 [`wait_timeout`](/system-variables.md#wait_timeout) 和 [`interactive_timeout`](/system-variables.md#interactive_timeout) 两种超时。
+TiDB 目前支持 [`wait_timeout`](/system-variables.md#wait_timeout)，[`interactive_timeout`](/system-variables.md#interactive_timeout) 和 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout) 三种超时。
```
Comment (Oreoxmt) [inline faq/manage-cluster-faq.md:66]:
```suggestion
TiDB 目前支持 [`wait_timeout`](/system-variables.md#wait_timeout)、[`interactive_timeout`](/system-variables.md#interactive_timeout) 和 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout) 三种超时。
```

**Review by Oreoxmt — APPROVED** [review]

File: faq/manage-cluster-faq.md [hunk 6]
```diff
@@ -63,7 +63,7 @@ mysql -h 127.0.0.1 -uroot -P4000
 
 ### TiDB 是否支持会话超时？
 
-TiDB 目前支持 [`wait_timeout`](/system-variables.md#wait_timeout) 和 [`interactive_timeout`](/system-variables.md#interactive_timeout) 两种超时。
+TiDB 目前支持 [`wait_timeout`](/system-variables.md#wait_timeout)、[`interactive_timeout`](/system-variables.md#interactive_timeout) 和 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout) 三种超时。
```
Comment (Oreoxmt) [inline faq/manage-cluster-faq.md:66]:
```suggestion
TiDB 目前支持 [`wait_timeout`](/system-variables.md#wait_timeout)、[`interactive_timeout`](/system-variables.md#interactive_timeout) 和 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout-从-v760-版本开始引入) 三种超时。
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 7]
```diff
@@ -20,6 +20,8 @@ TiDB 的事务的实现采用了 MVCC（多版本并发控制）机制，当新
 
 ## 事务超时
 
+在事务已启动但未提交或回滚的情况下，可能需要更细粒度的控制和更短的超时，以避免持有锁的时间过长。TiDB 在 v7.6.0 引入了 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout) 用来控制用户会话中事务的空闲超时。
```
Comment (Oreoxmt) [inline develop/dev-guide-timeouts-in-tidb.md:23]:
```suggestion
在事务已启动但未提交或回滚的情况下，可能需要更细粒度的控制和更短的超时，以避免持有锁的时间过长。TiDB 在 v7.6.0 引入了 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout-从-v760-版本开始引入) 用来控制用户会话中事务的空闲超时。
```

File: develop/dev-guide-connection-parameters.md [hunk 8]
```diff
@@ -239,4 +239,6 @@ UPDATE `t` SET `a` = 12 WHERE `id` = 3;
 
 TiDB 提供两个与 MySQL 兼容的超时控制参数，[`wait_timeout`](/system-variables.md#wait_timeout) 和 [`max_execution_time`](/system-variables.md#max_execution_time)。这两个参数分别控制与 Java 应用连接的空闲超时时间和连接中 SQL 执行的超时时间，即控制 TiDB 与 Java 应用的连接最长闲多久和最长忙多久。在 TiDB v5.4 及以上版本中，`wait_timeout` 参数默认值为 `28800` 秒，即空闲超时为 8 小时。在 v5.4 之前，`wait_timeout` 参数的默认值为 `0`，即没有时间限制。 `max_execution_time` 参数的默认值为 `0`，即不限制一条 SQL 语句的执行时间。
 
+但是 [`wait_timeout`](/system-variables.md#wait_timeout) 的默认值比较大，在事务已启动但未提交或回滚的情况下，可能需要更细粒度的控制和更短的超时，以避免持有锁的时间过长。TiDB 在 v7.6.0 引入了 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout) 用来控制用户会话中事务的空闲超时。
```
Comment (Oreoxmt) [inline develop/dev-guide-connection-parameters.md:242]:
```suggestion
但是 [`wait_timeout`](/system-variables.md#wait_timeout) 的默认值比较大，在事务已启动但未提交或回滚的情况下，可能需要更细粒度的控制和更短的超时，以避免持有锁的时间过长。TiDB 在 v7.6.0 引入了 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout-从-v760-版本开始引入) 用来控制用户会话中事务的空闲超时。
```

**Review by benmaoer — APPROVED** [review]
LGTM

File: system-variables.md [hunk 9]
```diff
@@ -2666,6 +2666,18 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 31536000]`
+- 单位：秒
+- 这个变量用来控制用户会话中事务的空闲超时。当用户会话处于事务状态且空闲时间超过该变量设定的值时，会话会被 Kill 掉。用户会话空闲是指没有正在执行的请求，处于等待请求的状态。
```
Comment (qiancai) [inline system-variables.md:2678]:
```suggestion
- 这个变量用来控制用户会话中事务的空闲超时。当用户会话处于事务状态且空闲时间超过该变量设定的值时，会话会被 Kill 掉。事务的空闲是指事务没有正在执行的请求，处于等待请求的状态。
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 10]
```diff
@@ -20,6 +20,8 @@ TiDB 的事务的实现采用了 MVCC（多版本并发控制）机制，当新
 
 ## 事务超时
 
+在事务已启动但未提交或回滚的情况下，可能需要更细粒度的控制和更短的超时，以避免持有锁的时间过长。TiDB 在 v7.6.0 引入了 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout-从-v760-版本开始引入) 用来控制用户会话中事务的空闲超时。
```
Comment (qiancai) [inline develop/dev-guide-timeouts-in-tidb.md:23]:
```suggestion
在事务已启动但未提交或回滚的情况下，你可能需要更细粒度的控制和更短的超时，以避免持有锁的时间过长。此时，你可以使用 TiDB 在 v7.6.0 引入的 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout-从-v760-版本开始引入) 控制用户会话中事务的空闲超时。
```

File: develop/dev-guide-connection-parameters.md [hunk 11]
```diff
@@ -239,4 +239,6 @@ UPDATE `t` SET `a` = 12 WHERE `id` = 3;
 
 TiDB 提供两个与 MySQL 兼容的超时控制参数，[`wait_timeout`](/system-variables.md#wait_timeout) 和 [`max_execution_time`](/system-variables.md#max_execution_time)。这两个参数分别控制与 Java 应用连接的空闲超时时间和连接中 SQL 执行的超时时间，即控制 TiDB 与 Java 应用的连接最长闲多久和最长忙多久。在 TiDB v5.4 及以上版本中，`wait_timeout` 参数默认值为 `28800` 秒，即空闲超时为 8 小时。在 v5.4 之前，`wait_timeout` 参数的默认值为 `0`，即没有时间限制。 `max_execution_time` 参数的默认值为 `0`，即不限制一条 SQL 语句的执行时间。
 
+但是 [`wait_timeout`](/system-variables.md#wait_timeout) 的默认值比较大，在事务已启动但未提交或回滚的情况下，可能需要更细粒度的控制和更短的超时，以避免持有锁的时间过长。TiDB 在 v7.6.0 引入了 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout-从-v760-版本开始引入) 用来控制用户会话中事务的空闲超时。
```
Comment (qiancai) [inline develop/dev-guide-connection-parameters.md:242]:
```suggestion
但是 [`wait_timeout`](/system-variables.md#wait_timeout) 的默认值比较大，在事务已启动但未提交或回滚的情况下，你可能需要更细粒度的控制和更短的超时，以避免持有锁的时间过长。此时，你可以使用 TiDB 在 v7.6.0 引入的 [`tidb_idle_transaction_timeout`](/system-variables.md#tidb_idle_transaction_timeout-从-v760-版本开始引入) 控制用户会话中事务的空闲超时。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: system-variables.md (hunk 9 above)
Comment (crazycs520) [reply system-variables.md:2678]: 还是原来的语义好些。

**Pull request merged**